### DISCUSSION TITLE
Change response content based on request URI

### DISCUSSION THREAD

**Discussion by atcasanova** [synfron/ReshaperForBurp#35] [General]
I'm simply dying to do it.
I want to change the JSON content of response for a specific request.
This is what i'm doing:

Whens:
<img width="341" alt="image" src="https://user-images.githubusercontent.com/2973929/228675860-3223fbbb-0efc-4627-8978-c6c587f063df.png">
and
<img width="342" alt="image" src="https://user-images.githubusercontent.com/2973929/228675956-25a95934-0967-4101-90bd-d57c8beab546.png">

Thens:
highlight green, set encoding UTF-8 and:

<img width="343" alt="image" src="https://user-images.githubusercontent.com/2973929/228676073-511b4146-35eb-46c4-bbcb-95fc6e0ef537.png">

I've tried some slightly different approaches, but i'm having no success. Best I could get was an empty answer.

Any ideas?

**Comment by ddwightx** [comment]
@atcasanova The main issue I see is that When > Event Direction is set to `Request`. Instead, that should be `Response` given you want to change the incoming response. There should be a message in the Log tab warning you of this issue.

Other than that I see your Then actions should work but they aren't optimal:
1. There isn't a need to set the encoding unless your JSON has non-ASCII characters in it. So you probably can delete that one.
2. Given you are setting the whole response body, there is no need to use replace. Uncheck `Use Message Value`  and set `Source Text` to be your JSON. Then uncheck  `Use Regex Replace`.

**Reply by atcasanova** [reply]
actually i need to change the response based only in the request URI. It's the only way to single out the response i need to change. As for the response encoding, it will 100% have non-ascii characters, that's why I forced the encoding to UTF-8.
I'll try your suggestions, and thanks for the quick reply.

**Reply by atcasanova** [reply]
It definetely worked! Thansk, @ddwightx !
